geowave-vector/src/main/java/mil/nga/giat/geowave/vector/util/FeatureDataUtils.java

```diff
@@ -0,0 +1,79 @@
+package mil.nga.giat.geowave.vector.util;
+
+import mil.nga.giat.geowave.vector.plugin.GeoWaveGTDataStore;
+
+import org.apache.log4j.Logger;
+import org.geotools.feature.simple.SimpleFeatureBuilder;
+import org.geotools.geometry.jts.JTS;
+import org.geotools.referencing.CRS;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.geometry.MismatchedDimensionException;
+import org.opengis.referencing.FactoryException;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+import org.opengis.referencing.operation.MathTransform;
+import org.opengis.referencing.operation.TransformException;
+
+import com.vividsolutions.jts.geom.Geometry;
+
+public class FeatureDataUtils
+{
+	private final static Logger LOGGER = Logger.getLogger(FeatureDataUtils.class);
+
+	public static SimpleFeature defaultCRSTransform(
+			final SimpleFeature entry,
+			final SimpleFeatureType persistedType,
+			final SimpleFeatureType reprojectedType,
+			final MathTransform transform ) {
+
+		// if the feature is in a different coordinate reference system than
+		// EPSG:4326, transform the geometry
+		final CoordinateReferenceSystem crs = entry.getFeatureType().getCoordinateReferenceSystem();
+		SimpleFeature defaultCRSEntry = entry;
+
+		if (!GeoWaveGTDataStore.DEFAULT_CRS.equals(crs)) {
+			MathTransform featureTransform = null;
+			if ((persistedType.getCoordinateReferenceSystem() != null) && persistedType.getCoordinateReferenceSystem().equals(
+					crs) && (transform != null)) {
+				// we can use the transform we have already calculated for this
+				// feature
+				featureTransform = transform;
+			}
+			else if (crs != null) {
+				// this feature differs from the persisted type in CRS,
+				// calculate the transform
+				try {
+					featureTransform = CRS.findMathTransform(
+							crs,
+							GeoWaveGTDataStore.DEFAULT_CRS,
+							true);
+				}
+				catch (final FactoryException e) {
+					LOGGER.warn(
```

Comment:
Comment only helps if people read it.  Whoops.   
New ticket is fine - I can make one